### PR TITLE
Update examples to use definitions.py (part 1)

### PR DESCRIPTION
**Pull request by maximearmstrong**
## Summary & Motivation

After discussion with the team, we will update our examples, docs and Dagster University courses to recommend initializing the Definitions object in `definitions.py` instead of `__init__.py`.

This PR is a first effort to update our examples. 

Examples that are missing from this PR and should be updated in another one:
- assets_*
- experimental
- project_*
- tutorial_*

Examples that we should be updated when we update the docs:
- doc_snippets

## How I Tested These Changes

Each example was tested locally + BK

### CODE REVIEW

**Comment by maximearmstrong** [comment]
* **#20771** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/20771?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @maximearmstrong and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by yuhan — APPROVED** [review]
i believe all the projects have test_defs unit tests so this should be all good 👍

File: examples/feature_graph_backed_assets/feature_graph_backed_assets/definitions.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+
+defs = Definitions(
+    assets=[
+        *load_assets_from_package_module(assets),
+        AssetsDefinition.from_graph(us_assets),
+        AssetsDefinition.from_graph(layover_breakdown_2022),
+    ],
+    jobs=[define_asset_job("airline_job", AssetSelection.assets("passenger_flights").downstream())],
+)
```
Comment (schrockn) [inline examples/feature_graph_backed_assets/feature_graph_backed_assets/definitions.py:24]: @maximearmstrong pre-existing problem but `airline_job` variable appears unused and pointless?

Comment (maximearmstrong) [reply examples/feature_graph_backed_assets/feature_graph_backed_assets/definitions.py:24]: Good catch - I missed that when moving things around. I will remove it in my second PR.

**Pull request merged**